fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccount.java

```diff
@@ -3713,14 +3713,14 @@ public Integer getSubStatus() {
     public void validateForAccountBlock() {
         final SavingsAccountSubStatusEnum currentSubStatus = SavingsAccountSubStatusEnum.fromInt(this.getSubStatus());
         if (SavingsAccountSubStatusEnum.BLOCK.hasStateOf(currentSubStatus)) {
-            throw new SavingsAccountBlockedException(this.getId());
+            throw new SavingsAccountBlockedException(getAccountNumber());
```

Comment:
This looks to me a breaking change.
Also I have checked and none of the other exceptions are using the account number! all of them using the account id instead…

What is the business need behind this change?

Reply:
Also please dont forget to add proper testing which validate your changes!